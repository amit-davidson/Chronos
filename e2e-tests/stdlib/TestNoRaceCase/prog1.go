package main

type Point struct {
	x, y int
}

func main() {
	var y int
	for x := -1; x <= 1; x++ {
		switch {
		case x < 0:
			y = -1
		case x == 0:
			y = 0
		case x > 0:
			y = 1
		}
	}
	y++
}
