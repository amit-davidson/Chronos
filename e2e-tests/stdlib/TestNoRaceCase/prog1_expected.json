{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"PosID": 1,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:11:10",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				}
			}
		},
		{
			"ID": 2,
			"PosID": 2,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:11:10",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 3,
			"PosID": 3,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:11:10",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				}
			}
		},
		{
			"ID": 4,
			"PosID": 4,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:19:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:8:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 5,
			"PosID": 5,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:19:2",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 5
				}
			}
		},
		{
			"ID": 6,
			"PosID": 6,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:17",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 7,
			"PosID": 7,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:17",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		},
		{
			"ID": 8,
			"PosID": 8,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:17",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				}
			}
		},
		{
			"ID": 9,
			"PosID": 9,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:23",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 9
				}
			}
		},
		{
			"ID": 10,
			"PosID": 10,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:23",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 10
				}
			}
		},
		{
			"ID": 11,
			"PosID": 11,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:13:10",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 11
				}
			}
		},
		{
			"ID": 12,
			"PosID": 12,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:13:10",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 12
				}
			}
		},
		{
			"ID": 13,
			"PosID": 13,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:13:10",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 13
				}
			}
		},
		{
			"ID": 14,
			"PosID": 14,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:15:10",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:9:6",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 14
				}
			}
		},
		{
			"ID": 15,
			"PosID": 15,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:15:10",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 15
				}
			}
		},
		{
			"ID": 16,
			"PosID": 16,
			"Pos": "",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/stdlib/TestNoRaceCase/prog1.go:15:10",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 16
				}
			}
		}
	]
}